Context/Application/Command/InternalBlockchain/DepositInternalContractRefund/DepositInternalContractRefundHandler.ts

```diff
@@ -0,0 +1,51 @@
+import {UseCase} from "context/Core/Domain/UseCase";
+import DepositInternalContractRefund from "./DepositInternalContractRefund";
+import DepositRepositoryInterface from "context/Domain/DepositRepositoryInterface";
+import InternalBlockchain from "context/InternalBlockchain/InternalBlockchain";
+import * as Errors from "./Errors"
+import {Inject, Injectable} from "@nestjs/common";
+
+@Injectable()
+export default class DepositInternalContractRefundHandler implements UseCase<DepositInternalContractRefund, void> {
+    constructor(
+        @Inject("DepositRepositoryInterface") private repository: DepositRepositoryInterface,
+        @Inject("InternalBlockchain") private internalBlockchain: InternalBlockchain
+    ) {}
+
+    async execute(command: DepositInternalContractRefund): Promise<void> {
+        const deposit = await this.repository.getById(command.depositId)
+
+        if (deposit === null) {
+
+            throw new Errors.DepositNotFound(command.depositId)
+        }
+
+        const internalContractId = deposit.internalContract?.internalId as string
+
+        if(!await this.hasRefundOperation(internalContractId, deposit._depositRequest.revpopAccount.value)) {
+            console.log(`Deposit ${deposit.idString} has not refund yet.`)
+
+            return;
+        }
+
+        console.log(`Deposit ${deposit.idString} has refund.`);
+
+        await this.internalBlockchain.burnAsset(parseInt(deposit._externalContract.value) / 100000000000000);
```

Comment:
Please make calculation more obvious (by defining a constant eg) and move it to the model.

Reply:
Moreover IMO, different assets might have different precisions.